demo/lib.py

```diff
@@ -81,35 +84,55 @@ def hash_claim(salt, value, return_raw=False):
     return hash_raw(raw.encode("utf-8"))
 
 
+def _create_sd_claim_entry(key, value:str, salt:str) -> str:
+    """
+     returns the hashed and salted value string
+     key arg is not used here, it's just for compliances to other calls
+    """
+    return hash_claim(salt, value)
+
+
+def _create_svc_entry(key, value:str, salt:str) -> str:
+    """
+     returns a string representation of a list 
+        [hashed and salted value string, value string]
+     key arg is not used here, it's just for compliances to other calls
+    """
+    return hash_claim(salt, value, return_raw=True)
+
+
 def create_sd_jwt_and_svc(user_claims, issuer, issuer_key, holder_key, claim_structure={}):
     """
     Create the SD-JWT
     """
-
+    # something like: {'sub': 'zyZQuxk2AUv5_Z_RAMxh9Q', 'given_name': 'EpCuoArhQK6MjmO6D-Bi6w' ...
     salts = walk_by_structure(
         claim_structure, user_claims, lambda _, __, ___=None: generate_salt()
     )
 
-    def _create_sd_claim_entry(_, value, salt):
-        return hash_claim(salt, value)
-
     # Create the JWS payload
     sd_jwt_payload = {
         "iss": issuer,
         "sub_jwk": holder_key.export_public(as_dict=True),
-        "iat": 1516239022,
-        "exp": 1516247022,
+        "iat": int(datetime.datetime.utcnow().timestamp()),
+        "exp": int(
+                    (
+                        datetime.datetime.now() + \
+                        datetime.timedelta(minutes=DEFAULT_EXP_MINS)
+                    ).utcnow().timestamp()
+        ),
```

Comment:
Can you please change this so it observes the "no randomness" mode (which would better be called "reduced randomness")? This way, we get less changes lines in git when the examples are recreated.

Reply:
Ok, so this code has also the scope to put examples in the specs

I'll put an argparse CLI with parameters to get static values or random one. Do you approve this approach?

Reply:
Yes! The code is rough around the edges and argparse should be in there anyway :-)